src/sentry/grouping/enhancer/__init__.py

```diff
@@ -129,6 +136,21 @@ def __init__(self, rules, version=None, bases=None, id=None):
             bases = []
         self.bases = bases
 
+        self.rust_enhancements = None
+        try:
+            merged_rust_enhancements = RustEnhancements.empty()
+            for base in bases:
+                base = ENHANCEMENT_BASES.get(base)
+                if base:
+                    if not base.rust_enhancements:
+                        raise Exception("base has no rust_enhancements")
```

Comment:
I factored the merging and parsing out into separate functions.
Originally, I wanted this to recursively fail if the "parent" failed parsing for whatever reason.
Now we just force parsing with Rust in the module load path (also to avoid any confusion related to `options` being available at module loading time), so we are guaranteed to always have the base rules available.